app/code/core/Mage/Widget/Model/Widget.php

```diff
@@ -89,7 +89,7 @@ public function getConfigAsObject($type)
         $object->setData($xml->asCanonicalArray());
 
         // Set module for translations etc.
-        $module = $object->getData('@/module');
+        $module = $xml->getAttribute('module');
```

Comment:
why is this line? I see the behavior is the same but it is for performance?

Reply:
This is actually why despite the good idea, Magento's whole translation helper implementation was broken.

A few lines above it calls `$xml->asCanonicalArray()`, which calls $xml->_asArray(true)`. Setting the parameter as true doesn't add attributes to the Varien_Object:

https://github.com/MahoCommerce/maho/blob/main/lib/Varien/Simplexml/Element.php#L210-L220

So $module was always null.

Reply:
I even went back with git blame on the OM repo to see if it worked at some point but then was broken, but nope it was always broken...